src/components/header/Header.js

```diff
@@ -1,52 +1,54 @@
-import { useState } from "react";
+import { useState, useEffect } from "react";
+// import DisplayImage from "../display/DisplayImage";
 import "./Header.css";
-import { FaSearch } from "react-icons/fa";
+// import { FaSearch } from "react-icons/fa";
 import ImageForm from "../form/ImageForm";
 import axiosInstance from "../api/axios";
 
 const Header = () => {
   const [openModal, setOpenModal] = useState(false);
+  const [mount, setMount] = useState(false);
   const handleOpen = () => {
     setOpenModal(true);
   };
- 
+
   const [search, setSearch] = useState("");
 
   const handleSearch = async (e) => {
     e.preventDefault();
     setSearch(e.target.value);
   };
- 
-  
+
   const searchPictures = async () => {
     try {
-      const res = await axiosInstance.get(`${search}`);
-      setSearch(res.data.results);
+       await axiosInstance.get(`${search}`);
+      setSearch(search);
     } catch (error) {
-      console.log(error)
+      console.log(error);
     }
   };
-  
-  useEffect(() => {
-    searchPictures();
-  },[]);
 
 
+  useEffect(() => {
+    if (!mount) {
+      setMount(true);
+      searchPictures();
+    }
+  }, [searchPictures, mount]);
+
```

Comment:
The `searchPictures` declaration should be inside this `useEffect`

Reply:
I now used without the useEffect, now works perfectly without re-rendering issue.